docs/source/protocol.rst

```diff
@@ -155,107 +146,41 @@ Event ID
 - Listed in :ref:`events page`
 - See: http://linux.die.net/man/3/uuid
 
-Count
-+++++
-
-- 64-bit signed integer
-- GVariant symbol: ``x``
-
-Relative Timestamp
-++++++++++++++++++
-
-- 64-bit signed integer
-- GVariant symbol: ``x``
-- See: http://linux.die.net/man/3/clock_gettime
+OS Version
+++++++++++
 
-Auxiliary Payload
-+++++++++++++++++
+- String
+- GVariant symbol: ``s``
 
-- Maybe Variant
-- Allows for no contents ``(NULL)`` or content of any type.
-- Used to contain data associated with an event that is logged.
-- GVariant symbol: ``mv``
-- See: https://developer.gnome.org/glib/stable/gvariant-format-strings.html#gvariant-format-strings-maybe-types
-- Details for each event ID listed in :ref:`events page`
 
-Total format
+Period Start
 ++++++++++++
 
-In total should look like ``(uayxxmv)``.
-
-Oddities
-++++++++
-
-Aggregates currently have no starting time, only a stopping time. Will be fixed
-by `T8261: Metrics Client: Aggregates record start time <https://phabricator.endlessm.com/T8261>`_
-
-Sequence Metric
-~~~~~~~~~~~~~~~
-
-Sequence metrics are a type of metric that has elements spread across time,
-sequentially (hence the name!) All sequence metrics have a **start** event,
-zero or more **progress** event, and terminate with a **stop** event.
-
-Sequences Metrics' events are chronologically ordered. It should not be
-possible for events in an event sequence to arrive out-of-order
-chronologically.
-
-In order:
-
-User ID
-+++++++
-
-- Unsigned 32-bit integer
-- GVariant symbol ``u``
-- See: http://man7.org/linux/man-pages/man2/getuid.2.html
-
-Event ID
-++++++++
-
-- Array of 16 unsigned bytes
-- GVariant symbol ``ay``
-- Listed in :ref:`events page`
-- See: http://linux.die.net/man/3/uuid
-
-Events
-++++++
+- String
+- GVariant symbol: ``s``
+- Date string format: ``YYYY-MM-DD``
+- Beginning of the period, with aggregation done using user’s computer time
 
-- Array of relative timestamps and auxiliary payloads.
-- First element in array is a start event.
-- Last element in array is a stop event.
-- Any elements in between are progress events.
-- GVariant symbol ``a(xmv)``
-
-Relative Timestamp
-++++++++++++++++++
+Count
++++++
 
 - 64-bit signed integer
-- GVariant symbol ``x``
-- See: http://linux.die.net/man/3/clock_gettime
+- GVariant symbol: ``x``
```

Comment:
Above we document that this is non-negative. So… it should either be a 64-bit **un**signed int. Or, if we are avoiding that type because postgresql only has signed 64-bit ints, we should explain this discrepancy somewhere.

Reply:
Right. I’ve transformed this into a 32-bit unsigned integer, it should be big enough and has no problem with PostgreSQL. Downside: it changes the GVariant symbol.